classes/utils/Utils.py

```diff
@@ -0,0 +1,321 @@
+import re
+
+from enum import Enum, IntEnum
+
+
+class Version(Enum):
+
+    def __init__(self, major, minor, version, protocol):
+        self.major = major
+        self.minor = minor
+        self.version = version
+        self.protocol = protocol
+        self.all = (major, minor, version)
+
+    def is_same_major(self, other_version):
+        return self.major == other_version.major
+
+    def get_version(protocol):
+        for v in Version:
+            if v.protocol == protocol:
+                return v
+        return None
+
+    def __lt__(self, other):
+        if type(other) == tuple:
+            return self.all < other
+        else:
+            return self.all < other.all
+
+    def __le__(self, other):
+        return not self.__gt__(other)
+
+    def __eq__(self, other):
+        if type(other) == tuple:
+            return self.all == other
+        else:
+            return self.all == other.all
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __ge__(self, other):
+        return not self.__lt__(other)
+
+    def __gt__(self, other):
+        if type(other) == tuple:
+            return self.all > other
+        else:
+            return self.all > other.all
+
+    v1_12_0 = (1, 12, 0, 335)
+    v1_12_1 = (1, 12, 1, 338)
+    v1_12_2 = (1, 12, 2, 340)
+    v1_13_0 = (1, 13, 0, 393)
+    v1_13_1 = (1, 13, 1, 401)
+    v1_13_2 = (1, 13, 2, 404)
+    v1_14_0 = (1, 14, 0, 477)
+    v1_14_1 = (1, 14, 1, 480)
+    v1_14_2 = (1, 14, 2, 485)
+    v1_14_3 = (1, 14, 3, 490)
+    v1_14_4 = (1, 14, 4, 498)
+    v1_15_0 = (1, 15, 0, 573)
+    v1_15_1 = (1, 15, 1, 575)
+    v1_15_2 = (1, 15, 2, 578)
+    v1_16_0 = (1, 16, 0, 735)
+    v1_16_1 = (1, 16, 1, 736)
+    v1_16_2 = (1, 16, 2, 751)
+
+
+class GameMode(IntEnum):
+
+    SURVIVAL = 0
+    CREATIVE = 1
+    ADVENTURE = 2
+    SPECTATOR = 3
+
+
+regex_color = re.compile(r"§[0-9A-FK-OR]", re.IGNORECASE)
+
+
+class ChatColor(Enum):
+
+    def __init__(self, key, id):
+        self.key = key
+        self.id = id
+
+    def key():
+        return '§'
+
+    def strip_color(input):
+        return regex_color.sub('', input)
+
+    def translate_alternate_color(char, text):
+        # TODO
+        return text
+
+    BLACK = ('0', 0)
+    DARK_BLUE = ('1', 1)
+    DARK_GREEN = ('2', 2)
+    DARK_AQUA = ('3', 3)
+    DARK_RED = ('4', 4)
+    DARK_PURPLE = ('5', 5)
+    GOLD = ('6', 6)
+    GRAY = ('7', 7)
+    DARK_GRAY = ('8', 8)
+    BLUE = ('9', 9)
+    GREEN = ('a', 10)
+    AQUA = ('b', 11)
+    RED = ('c', 12)
+    LIGHT_PURPLE = ('d', 13)
+    YELLOW = ('e', 14)
+    WHITE = ('f', 15)
+    MAGIC = ('k', 16)
+    BOLD = ('l', 17)
+    STRIKETHROUGH = ('m', 18)
+    UNDERLINE = ('n', 19)
+    ITALIC = ('o', 20)
+    RESET = ('r', 21)
+
+
+class Dimension(Enum):
+
+    OVERWORLD = 1
+    NETHER = 2
+    THE_END = 3
```

Comment:
Any particular reason this isn't 0-indexed?
```suggestion
    CUSTOM = 0
    OVERWORLD = 1
    NETHER = 2
    THE_END = 3
```

Reply:
world IDs are:
0 = Overworld
-1 = Nether
1 = End